docs/guides/initialize-atom-on-render.mdx

```diff
@@ -93,3 +93,46 @@ This behavior is due to our child components looking for the lowest commmon `Pro
 For more information on `Provider` behavior, please read the docs [here](../core/provider.mdx).
 
 For more complex use cases, check out [Scope extension](../extensions/scope.mdx).
+
+
+### Using Typescript
+
+`useHydrateAtoms` has overloaded types and typescript cannot extract types from overloaded function. It is recommended to use a `Map` when passing initial atom values to the `useHydrateAtoms`.
+
+Here is a working example :
+
+```jsx
```

Comment:
```suggestion
```tsx
```

Reply:
No, I think it's jsx, all other documents are jsx